internal/services/subscription/data_source_location.go

```diff
@@ -0,0 +1,157 @@
+}
+
+type LocationZoneMapping struct {
+	LogicalZone  string `tfschema:"logical_zone"`
+	PhysicalZone string `tfschema:"physical_zone"`
+}
+
+func (r LocationDataSource) ResourceType() string {
+	return "azurerm_location"
+}
+
+func (r LocationDataSource) ModelObject() interface{} {
+	return &LocationDataSourceModel{}
+}
+
+func (r LocationDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"location": commonschema.LocationWithoutForceNew(),
+	}
+}
+
+func (r LocationDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"region_type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+		"zone_mappings": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"logical_zone": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+					"physical_zone": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r LocationDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Subscription.SubscriptionsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			id := commonids.NewSubscriptionID(subscriptionId)
+			resp, err := client.ListLocations(ctx, id, resourcesSubscription.DefaultListLocationsOperationOptions())
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return fmt.Errorf("%s was not found", id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", id, err)
+			}
+
+			if resp.Model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			if resp.Model.Value == nil {
+				return fmt.Errorf("retrieving %s: model value was nil", id)
+			}
+
+			var model LocationDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			normalizedLocation := location.Normalize(model.Location)
+
+			locationValue, err := getLocation(normalizedLocation, resp.Model.Value)
+			if err != nil {
+				return err
+			}
+
+			var state LocationDataSourceModel
+			state.ZoneMappings = flattenZonesMapping(locationValue)
+			state.RegionType = string(pointer.From(locationValue.Metadata.RegionType))
+			state.DisplayName = pointer.From(locationValue.DisplayName)
+			state.Location = normalizedLocation
+
+			metadata.ResourceData.SetId(fmt.Sprintf("%s/locations/%s", id.ID(), normalizedLocation))
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func getLocation(location string, input *[]resourcesSubscription.Location) (*resourcesSubscription.Location, error) {
+	for _, item := range *input {
+		if pointer.From(item.Name) == location {
```

Comment:
could we filter the location type to only get the physical location here using `location.Metadata.RegionType`?